src/main/java/com/google/sps/data/DeliverySlot.java

```diff
@@ -83,6 +83,23 @@ public DeliverySlot(Date startTime, Date endTime, String userId) throws BadReque
     this.userId = userId;
   }
 
+  /**
+   * Creates a delivery slot given a date, start and end times in miliseconds and the user id. If 
+   * canBeInThePast is set to true, the starting time can be past the current time. The boolean is
+   * necessary for creating and displaying the history of delivery slots.
+   */
+  public DeliverySlot(Date startTime, Date endTime, String userId, boolean canBeInThePast) throws BadRequestException {
+    if (startTime.compareTo(endTime) > 0) {
+      throw new BadRequestException("Please enter a valid time!");
+    }
+    if (!canBeInThePast && startTime.getTime() < System.currentTimeMillis()) {
```

Comment:
nit: Feels a bit weird you're telling the constructor whether to perform this validation or not.

Especially since you're duplicating the constructor above just for this. Consider having the caller verify whether this is in the past or not. There's nothing inherently wrong with a delivery slot in the past, it's just that it might not be appropriate in some contexts (creating one in the past). Hence the caller is in the best position to perform these validations.